src/lib/xmlparse.rs

```diff
@@ -846,10 +846,9 @@ pub struct XML_ParserStruct {
     pub m_freeTagList: *mut TAG,
     pub m_inheritedBindings: *mut BINDING,
     pub m_freeBindingList: *mut BINDING,
-    pub m_attsSize: c_int,
     pub m_nSpecifiedAtts: c_int,
     pub m_idAttIndex: c_int,
-    pub m_atts: *mut super::xmltok::ATTRIBUTE,
+    pub m_atts: Vec<[*const XML_Char; 2]>,
```

Comment:
What are the two pointers in this array? Is there a reason to use a 2 element array instead of a struct with field names here (or at least a type name and anonymous fields)?

Reply:
Looks like `repr(C) Attribute {name: *const XML_Char, value: *const XML_Char}` would be about right?

Reply:
Yep, exactly. I considered adding that structure, and wondered if it's worth adding a brand new structure just for this considering that the C code expects a contiguous array of pairs. I'm not against it though.

Reply:
I'm strongly in favor of adding a struct with names.